### PR TITLE
Add to cart

### PR DESCRIPTION
**Pull request by irajbhinde**
Files to check - 

- [ ] src/components/CartComponents/CartCard/BillCard.jsx
- [ ] src/components/CartComponents/CartCard/CartCard.jsx
- [ ] src/components/HomePage/HomePageCard/HomePageCard.jsx
- [ ] src/components/ProductListing/Cards/ProductCard.jsx
- [ ] src/components/WishListComponents/WishListCard/WishListCard.jsx
- [ ] src/components/context/cart-context.js
- [ ] src/components/reducer/cartReducer.jsx
- [ ] src/pages/home/HomePage.jsx

Added Features - 

1) Add to cart & add to wishlist
2) Additional categories
3) Cart page which includes items in cart & checkout card
4) Wishlist page
5) Nav which updates badges as per cart & wishlist items

To be implemented - AUTH & API Calls

GIFY - 
add_to_cart

### CODE REVIEW

File: src/components/CartComponents/CartCard/CartCard.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { useCart } from "../../context/cart-context";
+
+export default function CartCard({ product }) {
+  const { cartState, cartDispatch } = useCart();
+  const { wishlistProducts } = cartState;
+  console.log("PD", product);
```
Comment (tushargoyal812) [inline src/components/CartComponents/CartCard/CartCard.jsx]: remove console

File: src/components/ProductListing/Cards/ProductCard.jsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import { Link } from "react-router-dom";
+import { useCart } from "../../context/cart-context";
+
+export default function ProductCard({ product, key }) {
+  const { _id, title, image, subtitle, price, rating } = product;
+  const { cartState, cartDispatch } = useCart();
+  const { cartProducts, wishlistProducts } = cartState;
+  console.log("CP", cartProducts);
+  console.log("WP", wishlistProducts);
```
Comment (tushargoyal812) [inline src/components/ProductListing/Cards/ProductCard.jsx]: remove console

File: src/components/ProductListing/Cards/ProductCard.jsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+          </button>
+
+          <button
+            onClick={() =>
+              cartDispatch({ type: "ADD_TO_WISHLIST", payload: 1 })
+            }
+            className="bg-none btn-card vertical-card"
+          >
+            Add to Wishlist
+          </button>
```
Comment (tushargoyal812) [inline src/components/ProductListing/Cards/ProductCard.jsx]: remove commented code

File: src/components/WishListComponents/WishListCard/WishListCard.jsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import { Link } from "react-router-dom";
+
+import { useCart } from "../../context/cart-context";
+import "../WishListCard/wishListPage.css";
+
+export default function WishListCard({ product }) {
+  const { cartState, cartDispatch } = useCart();
+  const {wishlistProducts, cartProducts} = cartState
+  console.log("WL", product);
```
Comment (tushargoyal812) [inline src/components/WishListComponents/WishListCard/WishListCard.jsx]: remove console

**Review by tushargoyal812 — COMMENTED** [review]
code is clean and to the point.
naming convention looks good.

File: src/components/CartComponents/CartCard/BillCard.jsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { useCart } from "../../context/cart-context"
+
+export default function BillCard ({product}) {
+    const {cartState, cartDispatch} = useCart();
+    const {cartProducts} = cartState;
+    console.log("FOR PRICE", cartProducts);
```
Comment (ark2002) [inline src/components/CartComponents/CartCard/BillCard.jsx]: remove console before commits

File: src/components/CartComponents/cartNav/CartNav.jsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import { Link } from "react-router-dom";
+import { useCart } from "../../context/cart-context";
+
+export default function CartNav (){
+  const {cartState} = useCart();
+  const {cartProducts, wishlistProducts} = cartState
```
Comment (ark2002) [inline src/components/CartComponents/cartNav/CartNav.jsx:6]: nice work with destructuring

File: src/components/context/product-context.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    Anjunadeep: false,
+    Anjunabeats: false,
+    Above_And_Beyond: false,
+  },
+};
+
+const ProductContext = createContext(null);
+const useProduct = () => useContext(ProductContext);
+const ProductProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(productReducer, initialState);
```
Comment (ark2002) [inline src/components/context/product-context.js:19]: more specific names here would help if there are more reducers . the generic names can cause confusion

File: src/components/reducer/cartReducer.jsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+            }
+        case "REMOVE_FROM_WISHLIST":
+            return{
+                ...state,
+                wishlistProducts : [...state.wishlistProducts.filter((prod) => prod._id !== payload._id)]
+            }
+        case "ADD_TO_WISHLIST":
+            return{
+                ...state,
+                wishlistProducts : [...state.wishlistProducts, payload]
```
Comment (ark2002) [inline src/components/reducer/cartReducer.jsx:23]: dont need to spread if doing filter or map cause it makes a copy of the array anyway

File: src/components/reducer/productReducer.jsx [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+      return {
+        ...state,
+        sortByPrice: action.payload,
+      };
+    case "RATINGS":
+      return {
+        ...state,
+        sortByRating: action.payload,
+      };
+    case "ANJUNADEEP":
```
Comment (ark2002) [inline src/components/reducer/productReducer.jsx:14]: can destructure action into type and payload

File: src/components/utils/getSortByRating.jsx [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    }
+    if(sortByRating === "TWO_PLUS"){
+        return(
+            [...productList].filter((product) => product.rating > 2)
+        )
+    }
+    if(sortByRating === "ONE_PLUS"){
+        return(
+            [...productList].filter((product) => product.rating > 1)
+        )
```
Comment (ark2002) [inline src/components/utils/getSortByRating.jsx:22]: same with spreading here

**Review by ark2002 — APPROVED** [review]
all over code is easy to read and understand. just a small suggestion if you are using nav on all pages why make diffrent nav for each page declare the nav at the top level and keep it constant throughout the pages.

**Comment by ark2002** [comment]
also check your warnings in the deploy log you may have done some typo in importing the login page thats why the deploy preview has failed

image

**Pull request merged**